### PR TITLE
Implement "max PP" attribute in `BeatmapAttributeText`

### PR DESCRIPTION
**Pull request by cdwcgt**
- [ ] depends on  https://github.com/ppy/osu/pull/21585

related to #22096 

move `CalculatePerfectPerformance` logic to `PerformanceCalculator` for easy reuse, and I think it's its best place

https://user-images.githubusercontent.com/34775378/211538591-73c2b3a4-31ed-49d7-b289-16ceb379cada.mp4

Let me know if there is a place where I need to write a test, because I can't find a place to write a test

### CODE REVIEW

File: osu.Game/Skinning/Components/BeatmapAttributeText.cs
Comment (bdach) [inline osu.Game/Skinning/Components/BeatmapAttributeText.cs:65]: you could... you know, make one

File: osu.Game/Skinning/Components/BeatmapAttributeText.cs
Comment (bdach) [inline osu.Game/Skinning/Components/BeatmapAttributeText.cs:98]: handling mod changes is in scope of https://github.com/ppy/osu/pull/21585 and this pull does not adhere to the direction established therein (that both pre- and post-mod values should be available for use). so either this should be reverted and not handle mods, or this PR is hard-blocked until that one is merged

File: osu.Game/Skinning/Components/BeatmapAttributeText.cs
Comment (bdach) [inline osu.Game/Skinning/Components/BeatmapAttributeText.cs:136]: if the score is only manufactured to pass mods, then `CalculatePerfectPerformance` should just accept mods and not a score

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
it is my opinion that this PR should not and cannot proceed until we sort out `BeatmapAttributeText` properly. the initial implementation was very simple and very naive, but the component is seeing a lot of changes that attempt to tack expensive computations onto it. the component should be properly refactored to support this in a way that does not require all values to be recomputed whenever literally anything changes, and manages state changes locally and optimally.

**Comment by cdwcgt** [comment]
This PR is too old for now. need to a new pr to continue

**Pull request closed without merging**
